tests/nlu/dummy_data/dummy_features.py

```diff
@@ -0,0 +1,321 @@
+            attribute: the attribute for which concatenated features should be created
+            used_featurizers: the subset of the featurizers which should be used
+        Returns:
+            list containing, for each message, the concatenated features (i.e. one
+            matrix per per type and sparseness property)
+        """
+        used_featurizers_ordered = [
+            featurizer_description
+            for featurizer_description in self.featurizer_descriptions
+            if (used_featurizers is None)
+            or (featurizer_description.name in used_featurizers)
+        ]
+        collected: List[ConcatenatedFeaturizations] = []
+        for message in messages:
+            message_features: ConcatenatedFeaturizations = dict()
+            for is_sparse in [True, False]:
+                for feature_type in [FEATURE_TYPE_SEQUENCE, FEATURE_TYPE_SENTENCE]:
+                    sentences = feature_type == FEATURE_TYPE_SENTENCE
+                    matrices = [
+                        self.generate_pseudo_random_feature(
+                            attribute=attribute,
+                            tokens=message.get(TOKENS_NAMES[attribute], []),
+                            origin=fd.name,
+                            is_sparse=fd.is_sparse,
+                            feature_type=feature_type,
+                            feature_dim=(
+                                fd.sentence_dim if sentences else fd.sequence_dim
+                            ),
+                        ).features
+                        # concatenated only the "used featurizers"
+                        for fd in used_featurizers_ordered
+                        # concatenate only those with same sparseness property
+                        if fd.is_sparse == is_sparse
+                        # ... if attributes the attribute was featurized
+                        and attribute
+                        in (
+                            fd.sentence_attributes
+                            if sentences
+                            else fd.sequence_attributes
+                        )
+                        # ... if the attribute is tokenized
+                        and message.get(TOKENS_NAMES[attribute], [])
+                    ]
+                    if matrices:
+                        if is_sparse:
+                            matrix = scipy.sparse.hstack(matrices)
+                        else:
+                            matrix = np.concatenate(matrices, axis=-1)
+                        message_features[(feature_type, is_sparse)] = matrix
+            collected.append(message_features)
+        return collected
+
+    @staticmethod
+    def compare_with_feature_arrays(
+        actual: Dict[Text, List[FeatureArray]],
+        expected: List[ConcatenatedFeaturizations],
+    ) -> None:
+        """Compares concatenated features."""
+
+        for feature_type in [FEATURE_TYPE_SENTENCE, FEATURE_TYPE_SENTENCE]:
```

Comment:
I assume one of these is meant to be `SEQUENCE`?

Reply:
great catch :)  really hard to distinguish, if we ever rename I'd vote for something like `STATIC` vs `SEQUENTIAL` or so (differs in lengths and letters :D)